paperqa/types.py

```diff
@@ -570,17 +576,19 @@ def formatted_citation(self) -> str:
                 " to call `hydrate`?"
             )
 
-        quality = (
-            SOURCE_QUALITY_MESSAGES[self.source_quality]
-            if self.source_quality >= 0
-            else None
-        )
+        if self.source_quality_str:
+            return (
+                f"{self.citation} This article has {self.citation_count} citations and is"
```

Comment:
Does `self.citation` have a newline at the end of it?

Reply:
should not have that